lib/src/streams/deep_hash_stream.dart

```diff
@@ -0,0 +1,58 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'sha384_io.dart' if (dart.library.js) 'sha384_web.dart';
+
+Future<Uint8List> deepHashStream(dynamic data) async {
+  if (data is Stream<Uint8List>) {
+    final hasher = SHA384Hash();
+    await hasher.init();
+
+    int length = 0;
+
+    await for (final chunk in data) {
+      length += chunk.length;
+      hasher.update(chunk);
+    }
+
+    final tag = concatenateUint8Lists(
+        stringToUint8List('blob'), stringToUint8List(length.toString()));
+
+    final taggedHash =
+        concatenateUint8Lists(await sha384(tag), await hasher.hash());
+
+    return await sha384(taggedHash);
+  } else if (data is List<List> || data is List<Stream<Uint8List>>) {
+    final tag = concatenateUint8Lists(
+        stringToUint8List('list'), stringToUint8List(data.length.toString()));
+
+    return await deepHashChunks(data, await sha384(tag));
+  } else {
+    throw FormatException('Invalid data type.');
+  }
+}
+
+Future<Uint8List> deepHashChunks(dynamic chunks, Uint8List acc) async {
```

Comment:
`deepHashChunks` can be either a `Stream<Uint8List>`, `Uint8List`, a `List<List>` or `List<Stream<Uint8List>>`. As Dart doesn't have union types, using `dynamic` was the best option that I found :(

Do you have any suggestion?

Reply:
I was thinking of making the type be `Iterable<dynamic>` but I see `Stream` implements neither `List` nor `Iterable`.

What we can do is what you did for `deepHashStream`: to check if `data is Stream<Uint8List>` or `data is List<List> || data is List<Stream<Uint8List>>`, and to throw otherwise.

Having the method as it is (accessing properties we assume it has) is dangerous. 😬